03/app.js

```diff
@@ -0,0 +1,72 @@
+const imageContainer = document.querySelector(".images-list");
+const imageItem = document.querySelector(".images-list__item");
+const input = document.querySelector("input");
+
+const createGallery = function (e) {
+  const files = Array.from(e.target.files);
+  files.forEach(function (file) {
+    if (file && file.type.includes("image")) {
+      const newItem = imageItem.cloneNode(true);
+      newItem.classList.remove("images-list__item--prototype");
+      const newHeader = newItem.querySelector("header");
+      newHeader.innerText = file.name;
+      const newFooter = newItem.querySelector("footer");
+      newFooter.innerText = file.size.toFixed(2);
+
+      const reader = new FileReader();
+
+      reader.onload = function (readerEvent) {
+        const newImg = newItem.querySelector("img");
+        newImg.src = readerEvent.target.result;
+
+        imageContainer.appendChild(newItem);
+      };
+      reader.readAsDataURL(file);
+    }
+  });
+};
+
+/* /* nieudana proba refaktorowania kodu...
+nie moge uzyskac dostepu do
+
+const createNewItem = function () {
+  const newItem = imageItem.cloneNode(true);
+  newItem.classList.remove("images-list__item--prototype");
+  return newItem;
+};
+
+const createNewHeader = function (newItem) {
+  const newHeader = newItem.querySelector("header");
+  newHeader.innerText = file.name;
```

Comment:
Wszystkie wykorzystywane zmienne które nie mają zasięgu globalnego należy przekazać przez parametr - tutaj np. `file` jest taką zmienną.

Zwróć uwagę, że w JS istotne jest miejsce deklaracji funkcji, a nie miejsce jej wywołania.
Dlatego `file` nie jest dostępne.

Reply:
W przypadku tego zadania chodzi mi o zmienna newItem. W momencie kiedy tworze ja i modyfikuje w funkcji createNewItem() to pomimo ze ja zwracam nie moge uzyskac do niej dostepu w createNewHeader() ani w createNewFooter() a co za tym idzie i dalej w funkcji createGallery. Widze teraz ze i file powinno byc dodane jako parametr, ale nie doszlam do tego bledu jeszcze z powodu wlasnie zmiennej newItem.

Reply:
Jak robisz `return newItem` to wtedy przy wywołaniu musisz ją odebrać czyli `const newItem = createNewItem(); /* tutaj nazwa zmiennej nie jest istotna */`. Potem możesz ją przekazać jak to już robiłaś (zabrakło tej pierwszej części) tj. `createNewHeader(newItem); `

Reply:
Udalo sie! Chyba gdzies przegapilam ten temat albo zapomnialam. Chce tylko sie upewnic ze dobrze rozumiem. Deklaruje po raz kolejny zmienna o nazwie newItem (lub jakiejkolwiek innej nazwie) i po znaku rownosci wywoluje funkcje createNewItem() dzieki czemu mam wynik return tej funkcji zapisany w nowej zmiennej?

Reply:
Potwierdzam! Tak jest ;)